### PR TITLE
oauth timeout

### PR DESCRIPTION
**Pull request by earonesty**
add timeout

### CODE REVIEW

File: cloudsync/oauth/oauth_config.py [hunk 1]
```diff
@@ -118,6 +118,9 @@ def wait_auth(self, token_url, timeout=None, **kwargs):
                     raise OAuthError(self.failure_info)
                 raise OAuthError("Oauth interrupted")
 
+            if "timeout" not in kwargs:
```
Comment (reed-atakama) [inline cloudsync/oauth/oauth_config.py:121]: why not set the timeout default to 60?

Comment (reed-atakama) [reply cloudsync/oauth/oauth_config.py:121]: I guess someone could explicitly set it to None

Comment (howufiger7) [reply cloudsync/oauth/oauth_config.py:121]: not sure I follow... can you even have a named param and kw param with the same name? and are these meant to be different timeouts? if so, can they have different names?

Comment (earonesty) [reply cloudsync/oauth/oauth_config.py:121]: thats what this does

Comment (earonesty) [reply cloudsync/oauth/oauth_config.py]: ok i changed it to  be more clear.  the named param is a real timeout.   the timeout passed to the requests lib is an interpacket timeout that has no bearing on the real world notion of timeouts.

File: cloudsync/oauth/oauth_config.py [hunk 2]
```diff
@@ -109,7 +109,7 @@ def start_auth(self, auth_url, scope=None, **kwargs):
 
     def wait_auth(self, token_url, timeout=None, **kwargs):
```
Comment (reed-atakama) [inline cloudsync/oauth/oauth_config.py:110]: I think there was some confusion on the last commit in the PR. I think Ilya and I were inquiring about why the 'timeout' parameter in the function signature wasn't being used. IMO the timeout parameter should default to 60 and be passed into the OAuthToken call.

Comment (reed-atakama) [reply cloudsync/oauth/oauth_config.py:110]: Disregard. Just saw your comment in slack. I see the timeout parameter is global and has a different use case

**Pull request merged**